### DISCUSSION TITLE
Custom directive is not working

### DISCUSSION THREAD

**Discussion by jacksontong** [livewire/livewire#5651] [7. Bugs (archived)] [answered]
I wrote my own sortable plugin, it used to work before but now it doesn't work anymore, I found that the callback is not being called, even though the directive has been registered.

```html
<div wire:sortable>
    <h1>Sortable</h1>
</div>
<script>
    document.addEventListener('DOMContentLoaded', () => {
        Livewire.directive('sortable', (el) => {
            console.log('sortable', el); // no output on console
        });
    });
</script>
```
![image](https://user-images.githubusercontent.com/7871218/230821119-43c7ed5a-7731-4b96-975d-829953d9af81.png)

- Laravel 10.6.2 with Vite
- Livewire 2.12.3

**Comment by joshhanley** [comment]
@jacksontong I think the issue is Alpine.

Without Alpine included on the page, Livewire starts when the `DOMContentLoaded` event fires, but with Alpine, Alpine actually triggers the start.

So you could try removing Alpine to see if your directive now runs, and if that works, you could then change your directive event listener to listen for `alpine:init` instead of `DOMContentLoaded`.

Hope this helps!

**Reply by jacksontong** [reply]
@joshhanley Thanks for looking at this, but I'm afraid it's not working, I've created a test repo here https://github.com/jacksontong/livewire-directive-test/blob/main/resources/views/dashboard.blade.php#L26

even if I remove alpine the code is still not working.

**Reply by joshhanley** [reply, marked as answer]
@jacksontong I had a look at your repo and the issue was that you don't have `wire:sortable` inside a Livewire component.

So I made a `Dashboard` component and copied your `dashboard.blade.php` into it (except for the `<x-app-layout>` tags). I then changed the dashboard route to point to it directly `Route::get('/dashboard', Dashboard::class)->name('dashboard');`, and it all worked happily.

Hope this helps!
